nautobot/core/celery/__init__.py

```diff
@@ -53,60 +43,74 @@ def register_task(self, task, **options):
 
 
 @signals.import_modules.connect
-def import_jobs_as_celery_tasks(sender, database_ready=True, **kwargs):
+def import_jobs(sender=None, **kwargs):
     """
-    Import system Jobs into Celery as well as Jobs from JOBS_ROOT and GIT_ROOT.
+    Import system Jobs into Nautobot as well as Jobs from JOBS_ROOT and GIT_ROOT.
+
+    Note that app-provided jobs are automatically imported at startup time via NautobotAppConfig.ready()
+    """
+    import nautobot.core.jobs  # noqa: F401
+
+    _import_jobs_from_jobs_root()
 
-    Note that app-provided Jobs are automatically imported at startup time via NautobotAppConfig.ready()
+    try:
+        _import_jobs_from_git_repositories()
+    except ProgrammingError:  # Database not ready yet, as may be the case on initial startup and migration
+        pass
```

Comment:
This feels gross but it does seem to work and makes it possible for this function to run from the Nautobot main server process even before migrations have been completed.

Reply:
Does the `database_ready` kwarg not work anymore to avoid having to do this?

Reply:
IIRC it didn't work exactly to our needs, for similar reasons to why we had to invent the `nautobot_database_ready` signal - if I remember what I encountered correctly, the issue is that during first-time startup of a fresh install, `AppConfig.ready()` gets called before migrations run (so it'd be `database_ready=False`) , but the same function gets called (with the same args) when restarting Nautobot in a steady-state scenario (so it'd be `database_ready=True`). Django doesn't provide any obvious flag that I've ever been able to find to readily distinguish between these two states, so we either need a try/except pattern like here or else we'd need to call into the Django migration code to actually query whether migrations have completed. 

We can't use the `nautobot_database_ready` signal to load jobs into memory because that only gets called specifically in response to `migrate` or `post_upgrade` commands and we need to load jobs into memory on normal server startup.